app/src/main/java/ch/hikemate/app/model/authentication/AuthViewModel.kt

```diff
@@ -50,10 +50,22 @@ class AuthViewModel(private val repository: AuthRepository) : ViewModel() {
    * @param email The email address of the user.
    * @param password The password of the user.
    */
-  fun createAccountWithEmailAndPassword(email: String, password: String) {
+  fun createAccountWithEmailAndPassword(
+      email: String,
+      password: String,
+      onSuccess: () -> Unit,
+      onErrorAction: (Exception) -> Unit
+  ) {
+    if (email.isEmpty() || password.isEmpty()) {
+      onErrorAction(Exception("Email and password must not be empty"))
+      return
+    }
```

Comment:
Nitpick: It's a great check to have here, but why don't you handle it **also** in the View? You already handle passwords not matching, and I think it would be cleaner to check for this issue too (with an additional check here!). It looks a bit "late" to me to wait for the ViewModel to throw an error like this, since it can be handled by the View too

Reply:
I did this as the firebase function will throw an error if given empty credentials.
I added a check in the view